zhaquirks/tuya/ts0601_din_power.py

```diff
@@ -177,6 +183,78 @@ def _update_attribute(self, attrid, value):
             self.endpoint.electrical_measurement.power_factor_reported(value / 10)
 
 
+class ZemismartManufCluster(TuyaManufClusterAttributes):
+    """Manufacturer Specific Cluster of the Zemismart SPM01 Power Meter device."""
+
+    attributes = {
+        ZEMISMART_TOTAL_ENERGY_ATTR: ("energy", t.uint32_t, True),
+        ZEMISMART_TOTAL_REVERSE_ENERGY_ATTR: ("reverse_energy", t.uint32_t, True),
+        ZEMISMART_VCP_ATTR: ("vcp_raw", t.data64, True),
+    }
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == ZEMISMART_TOTAL_ENERGY_ATTR:
+            self.endpoint.smartenergy_metering.energy_deliver_reported(value)
+        elif attrid == ZEMISMART_TOTAL_REVERSE_ENERGY_ATTR:
+            self.endpoint.smartenergy_metering.energy_receive_reported(value)
+        elif attrid == ZEMISMART_VCP_ATTR:
+            self.endpoint.electrical_measurement.voltage_reported(
+                (value[7] * 256) + value[6]
+            )
+            self.endpoint.electrical_measurement.current_reported(
+                (value[5] * 256 * 256) + (value[4] * 256) + value[3]
+            )
+            self.endpoint.electrical_measurement.power_reported(
+                (value[2] * 256 * 256) + (value[1] * 256) + value[0]
+            )
+
+
+class ZemismartPowerMeasurement(LocalDataCluster, ElectricalMeasurement):
+    """Custom class for power, voltage and current measurement."""
+
+    cluster_id = ElectricalMeasurement.cluster_id
+
+    POWER_ID = 0x050B
+    VOLTAGE_ID = 0x0505
+    CURRENT_ID = 0x0508
+
+    AC_VOLTAGE_MULTIPLIER = 0x0600
+    AC_VOLTAGE_DIVISOR = 0x0601
+    AC_CURRENT_MULTIPLIER = 0x0602
+    AC_CURRENT_DIVISOR = 0x0603
+
+    _CONSTANT_ATTRIBUTES = {
+        AC_VOLTAGE_MULTIPLIER: 1,
+        AC_VOLTAGE_DIVISOR: 10,
+        AC_CURRENT_MULTIPLIER: 1,
+        AC_CURRENT_DIVISOR: 1000,
+    }
+
+    def voltage_reported(self, value):
+        """Voltage reported."""
+        self._update_attribute(self.VOLTAGE_ID, value)
+
+    def power_reported(self, value):
+        """Power reported."""
+        self._update_attribute(self.POWER_ID, value)
+
+    def current_reported(self, value):
+        """Ampers reported."""
+        self._update_attribute(self.CURRENT_ID, value)
```

Comment:
Since this is relatively simple, we can call `update_attribute` on this cluster from another cluster and remove these methods.